lib/galaxy/config/sample/datatypes_conf.xml.sample

```diff
@@ -247,6 +247,7 @@
     </datatype>
     <datatype extension="directory" type="galaxy.datatypes.data:Directory">
     </datatype>
+    <datatype extension="yaml" type="galaxy.datatypes.text:Yaml" display_in_upload="true" />
```

Comment:
If this list is what users search through to assign datatypes in the UI, we should also include the `yml` extension here.

Reply:
Good point. I'll add it in the last commit. Thanks!

Reply:
I don't think that's desirable, this will require a datatype subclass and will unnecessarily inflate the (already long) datatype list.

Reply:
I agree with @nsoranzo . But a short description would be nice.

Reply:
xref: https://yaml.org/faq.html

Reply:
Is there not a way to have 2 extensions link to a single datatype? I'm not familiar with the intricacies of datatypes here, and having had a brief look, I don't understand how the subclasses work... but in this case it probably wouldn't matter if a `.yml` file was converted to `Yaml` for all internal/tool uses.

`.yml` may not be officially encouraged by the YAML developers, but it is still widely used – and I would guess that most users of YAML files don't know which is the preferred extension. I've been using YAML for a couple of years and I only discovered `.yaml` is preferred when I looked it up before commmenting this yesterday.

I guess whether this is appropriate depends on whether users believe they search datatypes by file _extension_ or file _type_. So far I've only searched by extension, so I would naturally search for `yml` when uploading a `.yml` file, and I might assume YAML wasn't supported if I didn't find any result for that.

Reply:
With the availability of a sniffer I would hope that users would only need to choose the datatype manually in corner cases.